TYPO3.Flow/Classes/TYPO3/Flow/Object/Proxy/ObjectSerializationTrait.php

```diff
@@ -0,0 +1,155 @@
+            if (!property_exists($this, 'Flow_Persistence_RelatedEntities') || !is_array($this->Flow_Persistence_RelatedEntities)) {
+                $this->Flow_Persistence_RelatedEntities = [];
+                $this->Flow_Object_PropertiesToSerialize[] = 'Flow_Persistence_RelatedEntities';
+            }
+            if ($propertyValue instanceof \Doctrine\ORM\Proxy\Proxy) {
+                $className = get_parent_class($propertyValue);
+            } else {
+                $className = \TYPO3\Flow\Core\Bootstrap::$staticObjectManager->getObjectNameByClassName(get_class($propertyValue));
+            }
+            $identifier = \TYPO3\Flow\Core\Bootstrap::$staticObjectManager->get('TYPO3\Flow\Persistence\PersistenceManagerInterface')->getIdentifierByObject($propertyValue);
+            if (!$identifier && $propertyValue instanceof \Doctrine\ORM\Proxy\Proxy) {
+                $identifier = current(\TYPO3\Flow\Reflection\ObjectAccess::getProperty($propertyValue, '_identifier', true));
+            }
+            $this->Flow_Persistence_RelatedEntities[$originalPropertyName . '.' . $path] = [
+                'propertyName' => $originalPropertyName,
+                'entityType' => $className,
+                'identifier' => $identifier,
+                'entityPath' => $path
+            ];
+            $this->$originalPropertyName = \TYPO3\Flow\Utility\Arrays::setValueByPath($this->$originalPropertyName, $path, null);
+        }
+    }
+
+    /**
+     * Reconstitues related entities to an unserialized object in __wakeup.
+     * Used in __wakeup methods of proxy classes.
+     *
+     * Note: This method adds code which ignores objects of type TYPO3\Flow\Resource\ResourcePointer in order to provide
+     * backwards compatibility data generated with Flow 2.2.x which still provided that class.
+     *
+     * @return void
+     */
+    private function Flow_setRelatedEntities()
+    {
+        if (property_exists($this, 'Flow_Persistence_RelatedEntities') && is_array($this->Flow_Persistence_RelatedEntities)) {
+            $persistenceManager = \TYPO3\Flow\Core\Bootstrap::$staticObjectManager->get('TYPO3\Flow\Persistence\PersistenceManagerInterface');
+            foreach ($this->Flow_Persistence_RelatedEntities as $entityInformation) {
+                if ($entityInformation['entityType'] === 'TYPO3\Flow\Resource\ResourcePointer') {
+                    continue;
+                }
+                $entity = $persistenceManager->getObjectByIdentifier($entityInformation['identifier'], $entityInformation['entityType'], true);
+                if (isset($entityInformation['entityPath'])) {
+                    $this->{$entityInformation['propertyName']} = \TYPO3\Flow\Utility\Arrays::setValueByPath($this->$entityInformation['propertyName'], $entityInformation['entityPath'], $entity);
```

Comment:
You forgot the second pair of curly brackets from https://github.com/neos/flow-development-collection/pull/63 
```
$this->{$entityInformation['propertyName']} = \TYPO3\Flow\Utility\Arrays::setValueByPath($this->$entityInformation['propertyName'], $entityInformation['entityPath'], $entity);
```
->
```
$this->{$entityInformation['propertyName']} = \TYPO3\Flow\Utility\Arrays::setValueByPath($this->{$entityInformation['propertyName']}, $entityInformation['entityPath'], $entity);
```

Without that this leads to a PHP Notice -> Exception